### DISCUSSION TITLE
eloquent computed attributes in laravel 9

### DISCUSSION THREAD

**Discussion by patrik-csak** [laravel/framework#42599] [Q&A]
the docs for laravel 8.x contain the following ([source](https://laravel.com/docs/8.x/eloquent-mutators#defining-an-accessor)) :

> You are not limited to interacting with a single attribute within your accessor. You may also use accessors to return new, computed values from existing attributes:
> 
>     /**
>      * Get the user's full name.
>      *
>      * @return string
>      */
>     public function getFullNameAttribute()
>     {
>         return "{$this->first_name} {$this->last_name}";
>     }

however, this section has been removed from the docs for laravel 9.x ([source](https://laravel.com/docs/9.x/eloquent-mutators#defining-an-accessor))

the following appears to work for me :

```php
<?php

namespace App\Models;
 
use Illuminate\Database\Eloquent\Casts\Attribute;
use Illuminate\Database\Eloquent\Model;
 
class User extends Model
{
    /**
     * Get the user's full name.
     *
     * @return \Illuminate\Database\Eloquent\Casts\Attribute
     */
    protected function fullName(): Attribute
    {
        return Attribute::make(
            get: fn () => "{$this->first_name} {$this->last_name}",
        );
    }
}
```

i'm wondering why this use-case has been removed from the docs and / or if there's another officially recommended way of implementing computed attributes

**Comment by liamseys** [comment]
@driesvints I'm wondering this too, can you elaborate the change? Thanks!

**Reply by driesvints** [reply]
Not sure sorry.

**Reply by Sophist-UK** [reply]
I would imagine that both old and new forms require searching through the reflection definitions to find functions that are either `get*Attribute()` or `*(): Attribute` to find the functions that need to be called whenever you create an actual model and need to create the actual attributes, so the performance is probably similar - or perhaps the new form only needs to be run once per model instantiation or maybe it is cacheable in some way - I haven't examined the code to see how these two syntaxes are handled in practice.

**Comment by pataar** [comment]
@driesvints Do you have any update on this? Is the old method (going to be) deprecated?

**Reply by driesvints** [reply]
It's not removed but it's recommended to use the new and documented way.

**Reply by pataar** [reply]
Thanks!
